Source/WebCore/Modules/filesystemaccess/FileSystemSyncAccessHandle.h

```diff
@@ -43,7 +43,7 @@ template<typename> class DOMPromiseDeferred;
 class FileSystemSyncAccessHandle : public ActiveDOMObject, public RefCounted<FileSystemSyncAccessHandle>, public CanMakeWeakPtr<FileSystemSyncAccessHandle> {
 public:
     struct FilesystemReadWriteOptions {
-        unsigned long long at;
+        std::optional<unsigned long long> at;
```

Comment:
@szewai are we supporting `std::optional<>` with dictionary?

Reply:
hmm I think `std::optional` is used with dictionary in FileSystemDirectoryHandle.idl?

Reply:
Yes, I think `FileSystemDirectoryHandle.idl`'s code is wrong since the generated code is,
```
  317 static inline JSC::EncodedJSValue jsFileSystemDirectoryHandlePrototypeFunction_removeEntryBody(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperationRet      urningPromise<JSFileSystemDirectoryHandle>::ClassParameter castedThis, Ref<DeferredPromise>&& promise)
  318 {
  319     auto& vm = JSC::getVM(lexicalGlobalObject);
  320     auto throwScope = DECLARE_THROW_SCOPE(vm);
  321     UNUSED_PARAM(throwScope);
  322     UNUSED_PARAM(callFrame);
  323     auto& impl = castedThis->wrapped();
  324     if (UNLIKELY(callFrame->argumentCount() < 1))
  325         return throwVMError(lexicalGlobalObject, throwScope, createNotEnoughArgumentsError(lexicalGlobalObject));
  326     EnsureStillAliveScope argument0 = callFrame->uncheckedArgument(0);
  327     auto name = convert<IDLUSVString>(*lexicalGlobalObject, argument0.value());
  328     RETURN_IF_EXCEPTION(throwScope, encodedJSValue());
  329     EnsureStillAliveScope argument1 = callFrame->argument(1);
  330     auto options = convert<IDLDictionary<FileSystemDirectoryHandle::RemoveOptions>>(*lexicalGlobalObject, argument1.value());
  331     RETURN_IF_EXCEPTION(throwScope, encodedJSValue());
  332     RELEASE_AND_RETURN(throwScope, JSValue::encode(toJS<IDLPromise<IDLUndefined>>(*lexicalGlobalObject, *castedThis->globalObject(), throwScope, [&]() -> decltype(auto) { return impl.remov      eEntry(WTFMove(name), WTFMove(options), WTFMove(promise)); })));
  333 }
```
```
  116 template<> FileSystemDirectoryHandle::RemoveOptions convertDictionary<FileSystemDirectoryHandle::RemoveOptions>(JSGlobalObject& lexicalGlobalObject, JSValue value)
  117 {
  118     VM& vm = JSC::getVM(&lexicalGlobalObject);
  119     auto throwScope = DECLARE_THROW_SCOPE(vm);
  120     bool isNullOrUndefined = value.isUndefinedOrNull();
  121     auto* object = isNullOrUndefined ? nullptr : value.getObject();
  122     if (UNLIKELY(!isNullOrUndefined && !object)) {
  123         throwTypeError(&lexicalGlobalObject, throwScope);
  124         return { };
  125     }
  126     FileSystemDirectoryHandle::RemoveOptions result;
  127     JSValue recursiveValue;
  128     if (isNullOrUndefined)
  129         recursiveValue = jsUndefined();
  130     else {
  131         recursiveValue = object->get(&lexicalGlobalObject, Identifier::fromString(vm, "recursive"_s));
  132         RETURN_IF_EXCEPTION(throwScope, { });
  133     }
  134     if (!recursiveValue.isUndefined()) {
  135         result.recursive = convert<IDLBoolean>(lexicalGlobalObject, recursiveValue);
  136         RETURN_IF_EXCEPTION(throwScope, { });
  137     } else
  138         result.recursive = false;
  139     return result;
  140 }
  ```
So, `auto options = convert<IDLDictionary<FileSystemDirectoryHandle::RemoveOptions>>` is not propagating optional information. And it is always creating a dictionary.

Reply:
ah I should correct that instead; thanks for pointing out